### PR TITLE
BUG/DOC: restrict possibilities for assembly of dates using a DataFrame

### PR DESCRIPTION
**Pull request by jreback**
xref #12967

### CODE REVIEW

**Comment by jreback** [comment]
cc @jorisvandenbossche

**Comment by jreback** [comment]
@shoyer

File: pandas/tseries/tools.py [hunk 1]
```diff
@@ -244,53 +244,10 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
...
-    >>> dfs = df.astype(str)
-    >>> pd.to_datetime(dfs.day + dfs.month + dfs.year, format="%d%m%Y")
-    0    2000-01-01
-    1    2000-01-02
-    ...
-    98   2000-04-08
-    99   2000-04-09
-    Length: 100, dtype: datetime64[ns]
-
-    Infer the format from the first entry
```
Comment (jorisvandenbossche) [inline pandas/tseries/tools.py]: I think this is still a useful example (the usage of infer_datetime_format), but if we keep it, it should just use a string column as input

Comment (jreback) [inline pandas/tseries/tools.py]: ok, i'll add that

**Comment by jorisvandenbossche** [comment]
Apart from the doc comment, I would personally even leave out the abbreviations for `minutes` and `seconds`. Then also the possible confusion of `m` as month or minute is avoided (I also never seen 'm' used as abbreviation of minute)

**Comment by shoyer** [comment]
+1 for leaving out any abbreviations. I'm not even sure it's actually a
good thing to accept both plural and singular forms.

**Pull request closed without merging**